src/connection.rs

```diff
@@ -4,15 +4,42 @@ use std::io::Write;
 use std::io::Read;
 use std::io;
 use std::net;
-
+#[cfg(unix)]
+use std::os;
+#[cfg(unix)]
+use std::convert::AsRef;
+#[cfg(unix)]
+use std::path::Path;
 use options::Options;
 use value::{ToMemcacheValue, FromMemcacheValue};
 use error::{MemcacheError, is_memcache_error};
 
 /// The connection acts as a TCP connection to the memcached server
 #[derive(Debug)]
-pub struct Connection {
-    reader: io::BufReader<net::TcpStream>,
+pub struct Connection<C: Read+Write+Sized> {
+    reader: io::BufReader<C>,
+}
```

Comment:
Need a blank line between each code block. You can just using the [rustfmt](https://github.com/rust-lang-nursery/rustfmt) to format all the codes.